python/grass/pygrass/vector/abstract.py

```diff
@@ -413,14 +413,33 @@ def open(
             str_err = "Not able to open the map, C function return %d."
             raise OpenError(str_err % openvect)
 
+        # Load attribute table for selected layer.
         if len(self.dblinks) == 0:
             self.layer = layer
             self.table = None
             self.n_lines = 0
         else:
-            self.layer = self.dblinks.by_layer(layer).layer
-            self.table = self.dblinks.by_layer(layer).table()
+            layer_db_link = self.dblinks.by_layer(layer)
+            if not layer_db_link:
+                raise LookupError(
+                    "There appears to be no database link for layer %d of <%s>."
+                    % (layer, self.name)
+                )
+            if layer_db_link.layer != layer:
+                raise RuntimeError(
+                    "The databse link for layer %d of <%s> references layer %d."
+                    % (layer, self.name, layer_db_link.layer)
+                )
+            self.layer = layer
+            try:
+                self.table = layer_db_link.table()
+            except Exception as error:
+                raise RuntimeError(
+                    "Loading the attribute table for layer %d of <%s> failed: %s"
+                    % (layer, self.name, error)
+                )
```

Comment:
In a hypothetical future where you require a recent Python version, this exception should not copy the error message but just `raise … from error`.

Reply:
The code on master branch is Python >=3.6, so if you don't want this backported to 7.8 or you prepare another PR for the 7.8 branch, you can do that already.

Reply:
Good to hear, I was judging from the `from __future__ import print_function` statement I saw. :wink:
I'll update the commit accordingly.

Reply:
Right, another PR needed there.